lua/neotest-haskell/init.lua

```diff
@@ -89,6 +102,9 @@ end
 
 setmetatable(HaskellNeotestAdapter, {
   __call = function(_, opts)
+    if opts.build_tools then
+      build_tools = opts.build_tools
+    end
```

Comment:
```suggestion
  __call = function(_, opts)
    vim.validate {
      build_tools = {
        opts.build_tools or build_tools,
        function(bts)
          if #bts == 0 then
            return false
          end
          for _, build_tool in bts do
            if not vim.tbl_contains(supported_build_tools, build_tool) then
              return false
            end
          end
          return true
        end,
        'at least one of ' .. table.concat(supported_build_tools, ', '),
      }

    }
    if opts.build_tools then
      build_tools = opts.build_tools
    end
```

...perhaps with the validation function extracted?

Reply:
I've added this. The error message is relatively hard to read for my setup, though. Still probably better to have that than an invalid configuration.